src/com/cft/contactmerge/io/tests/XmlImporterTest.java

```diff
@@ -1,18 +1,144 @@
+
+        assertThrows(IllegalArgumentException.class, () -> importer.Load(""));
+    }
+
+    @Test
+    void XmlImporter_Load_NonExistentFileName() {
+        // TODO: Would be great if we could test this without access the file system but articles
+        // I've read indicate you shouldn't try to mock File. How else to do it?
+        XmlImporter importer = new XmlImporter();
+
+        assertThrows(FileNotFoundException.class, () -> importer.Load("c:\\DoesNotExist\\DoesNotExist.xml"));
+    }
+
+    /* --------------------------------------------------------------------------------
+     * Testing Load(inputStream)
+     * -------------------------------------------------------------------------------- */
     @Test
-    void XmlImporter_Load() {
-        // TODO: Add tests
+    void XmlImporter_Load_NullStream() {
+        XmlImporter importer = new XmlImporter();
+
+        assertThrows(IllegalArgumentException.class, () -> importer.Load((InputStream) null));
     }
 
     @Test
-    void XmlImporter_Iterator() {
-        // TODO: Add tests
+    void XmlImporter_Load_EmptyStream() {
+        XmlImporter importer = new XmlImporter();
+
+        String testString = "";
+
+        InputStream inputStream = new ByteArrayInputStream(testString.getBytes());
+
+        assertThrows(UnknownFormatConversionException.class, () -> importer.Load(inputStream));
+    }
+
+    /* --------------------------------------------------------------------------------
+     * Testing Iterator
+     * -------------------------------------------------------------------------------- */
+
+    private static String testXml = "<SharpGridExport>\n" +
+            " <RowDefinition count=\"23\">\n" +
+            "  <Field alias=\"c1\" name=\"donor_donorid\" type=\"3\"/>\n" +
+            "  <Field alias=\"c2\" name=\"indiv_firstname\" type=\"8\"/>\n" +
+            "  <Field alias=\"c3\" name=\"indiv_lastname\" type=\"8\"/>\n" +
+            "  <Field alias=\"c4\" name=\"donor_phone\" type=\"8\"/>\n" +
+            "  <Field alias=\"c5\" name=\"donor_email\" type=\"8\"/>\n" +
+            "  <Field alias=\"c6\" name=\"donor_address1\" type=\"8\"/>\n" +
+            "  <Field alias=\"c7\" name=\"donor_address2\" type=\"8\"/>\n" +
+            "  <Field alias=\"c8\" name=\"donor_city\" type=\"8\"/>\n" +
+            "  <Field alias=\"c9\" name=\"donor_state\" type=\"8\"/>\n" +
+            "  <Field alias=\"c10\" name=\"donor_zip\" type=\"8\"/>\n" +
+            " </RowDefinition>\n" +
+            " <Data count=\"2\">\n" +
+            "  <row c1=\"13252\" c2=\"John\" c3=\"Doe\" c4=\"(520) 123-4567\" c5=\"jdoe@gmail.com\" c6=\"123 Main St\" c7=\"\" c8=\"Tucson\" c9=\"AZ\" c10=\"85750\" />\n" +
+            "  <row c1=\"13253\" c2=\"Adam\" c3=\"Smith\" c4=\"(520) 111-2222\" c5=\"adam.smith@yahoo.com\" c6=\"1010 Speedway Blvd\" c7=\"\" c8=\"Tucson\" c9=\"AZ\" c10=\"85750\" />\n" +
+            "</Data>\n" +
+            "</SharpGridExport>\n";
```

Comment:
Instead of hardcoding this, maybe actually creating an xml file, then importing it?

Reply:
Seems like a good rule and I can definitely understand future issues of having to manage test dependencies. I just want to ensure creating a test file isn't going to be like 2000 lines of code to read through. It may be a good idea to create a subClass containing methods that would contain test file contents?

I think it boils down to making the file contents easy to read through -vs- easy to manage both file contents and the test(s) that use them.